### PR TITLE
Revert recent NodeStateAdaptor changes (#87)

### PR DESCRIPTION
**Pull request by Anviking**
<p align="right">#</p>

# Overview

- [x] ~Reverted~Reset [`NodeStateAdaptor.hs`](https://github.com/input-output-hk/cardano-wallet/commits/develop/src/Cardano/Wallet/Kernel/NodeStateAdaptor.hs) to 1da9ca509d571fd18ef6e103795e7161dd697f26
- [x] Fixed errors (where we create it) until it compiled
- [x] Re-Added `retrying` for node api connection check

# Comments
This is not a complete revert of #87. There were several good things in there, like CLI changes, actually starting the node server, and cleanup.

- [ ] I should write a ticket if we're doing this

### CODE REVIEW

**Comment by Anviking** [comment]
(I’ll sanity check this tomorrow)

File: src/Cardano/Wallet/Server/Plugins.hs [hunk 1]
```diff
@@ -102,7 +104,7 @@ apiServer
     env <- ask
     let diffusion' = Kernel.fromDiffusion (lower env) diffusion
     logInfo "Testing node client connection"
-    eresp <- liftIO . NodeStateAdaptor.retrying . runExceptT $ NodeClient.getNodeSettings nodeClient
+    eresp <- liftIO . retrying . runExceptT $ NodeClient.getNodeSettings nodeClient
```
Comment (KtorZ) [inline src/Cardano/Wallet/Server/Plugins.hs:107]: What's the need for this now :thinking: ?

Comment (Anviking) [reply src/Cardano/Wallet/Server/Plugins.hs:107]: The [`node-settings`](https://github.com/input-output-hk/cardano-wallet/blob/171f15e4f4de0146f6d1acca70e3b5337cb396d0/src/Cardano/Wallet/API/V1/Handlers/Settings.hs#L18-L28) and `node-info` endpoints where we redirect to the node

File: src/Cardano/Wallet/Kernel/NodeStateAdaptor.hs [hunk 2]
```diff
@@ -259,6 +257,27 @@ data NodeStateAdaptor m = Adaptor {
...
+      -- in this context we need something very simple: a tuple containing the
+      -- "global blockchain height" and the "node blockchain height". The
+      -- former is the maximum between the biggest height we observed from an
+      -- unsolicited block we received and the current local tip:
+      --
+      -- global_height = max (last_known_header, node_local_tip)
+      --
+      -- The latter is simply the node local tip, i.e. "how far we went into
+      -- chasing the global blockchain height during syncing".
+    , getNodeSyncProgress :: LockContext -> m (Maybe BlockCount, BlockCount)
```
Comment (Anviking) [inline src/Cardano/Wallet/Kernel/NodeStateAdaptor.hs:279]: We could have `getNodeSyncProgress` remain removed, I guess, but ¯\_(ツ)_/¯

File: src/Cardano/Wallet/Server/Plugins.hs (hunk 1 above)
Comment (KtorZ) [reply src/Cardano/Wallet/Server/Plugins.hs:107]: I thought that was the goal of this PR to remove these calls :thinking: ?

Comment (Anviking) [reply src/Cardano/Wallet/Server/Plugins.hs:107]: Hm… I was more concerned about the internal uses of NodeStateAdaptor fields. Now these use `withNodeState` again.

Comment (KtorZ) [reply src/Cardano/Wallet/Server/Plugins.hs:107]: Okay. Fine by me.

**Review by KtorZ — APPROVED** [review]

**Pull request merged**